File: deno/src/lang/lang.graphql.ts
import { defineGraphql } from "/lib/context.ts";

import * as langResolver from "./lang.resolver.ts";

defineGraphql(langResolver, /* GraphQL */`
  
  type Query {
    "获取登录时的语言列表"
    getLoginLangs: [LangModel!]!
  }
  
`);

File: rust/src/src/base/lang/lang_graphql.rs
use anyhow::Result;
use async_graphql::{Context, Object};

use crate::common::context::Ctx;

use super::lang_resolver;
use crate::gen::base::lang::lang_model::LangModel;

#[derive(Default)]
pub struct LangQuery;

#[Object(rename_args = "snake_case")]
impl LangQuery {
  
  async fn get_login_langs(
    &self,
    ctx: &Context<'_>,
  ) -> Result<Vec<LangModel>> {
    let ctx = Ctx::builder(ctx)
      .build();
    
    let res = lang_resolver::get_login_langs(
      &ctx,
    ).await;
    
    ctx.ok(res).await
  }
  
}
